datalad/tests/utils.py

```diff
@@ -374,4 +374,18 @@ def newfunc(*arg, **kw):
             finally:
                 pass # might need to provide additional handling so, handle
     return newfunc
-with_testrepos.__test__ = False
\ No newline at end of file
+with_testrepos.__test__ = False
+
+def assert_cwd_unchanged(func):
```

Comment:
ideally, we should also get tests for the decorators... I haven't done it here yet but it would be great if you did for this -- it is a simple one to test

Thanks!